src/auto-evo/AutoEvoRun.cs

```diff
@@ -268,25 +270,29 @@ public void AddExternalPopulationEffect(Species species, int constant, float coe
     /// <returns>The summary of external effects.</returns>
     public LocalizedStringBuilder MakeSummaryOfExternalEffects()
     {
-        var combinedExternalEffects = new Dictionary<Tuple<Species, string>, long>();
+        var combinedExternalEffects = new Dictionary<(Species Species, string Event, Patch Patch), long>();
 
         foreach (var entry in ExternalEffects)
         {
-            var key = new Tuple<Species, string>(entry.Species, entry.EventType);
+            var key = (entry.Species, entry.EventType, entry.Patch);
+            var speciesPopulation = entry.Species.Population;
 
             combinedExternalEffects.TryGetValue(key, out var existingEffectAmount);
 
             combinedExternalEffects[key] = existingEffectAmount +
-                entry.Constant + (long)(entry.Species.Population * entry.Coefficient) - entry.Species.Population;
+                entry.Constant + (long)(speciesPopulation * entry.Coefficient) - speciesPopulation;
         }
 
         var builder = new LocalizedStringBuilder(300);
 
         foreach (var entry in combinedExternalEffects)
         {
-            // entry.Value is the amount, Item2 is the reason string
+            if (string.IsNullOrEmpty(entry.Key.Event))
```

Comment:
What's the case when the event is null or empty? Does this PR affect that or is this an unnecessary check?

Reply:
This is a bit of a hacky solution to avoid an event from showing up in the AutoEvo result. I didn't want to majorly rewrite this code just for the tiny edge case where after applying the population penalty for dying out in a patch the player has 0 global population which is invalid and might lead to bugs, therefore I summon some members to avoid this state: https://github.com/Revolutionary-Games/Thrive/blob/07932a5ac84e194cf45eaa94cacf98614e540e23/src/microbe_stage/MicrobeStage.cs#L391-L396

Reply:
Why? The case when the player is extinct is when they have 0 global population. So why is that required?

Reply:
This code runs when switching to a patch after the patch extinction screen. It's weird showing the player the game over screen (which would also require a specialized check) after telling him: hey, it's not over yet! You can still swich to another patch!

Reply:
Wouldn't it be better in that case to just  not show the extinction screen? I think that would be a cleaner design than adding a random population increase. Also this seems to give a pretty large population increase if the player was about to lose the game, in effect giving the player bunch of free lives if they managed to almost lose just right. So I'd say get rid of that, and just add some special flag in the stage that the player gets to respawn once more before extinction.

Reply:
I avoided this because I would consider it a bug being alive with 0 population. Also there are quite many places in the code where `Population > 0` is a check.

Reply:
In a new patch the player also has 0 population. And those checks for no population won't apply until the player goes to the editor at which point they gain the population bonus from that. So I do not see this as a problem at all. I find the randomly applied population bonus much more problematic design. If ever any player would noticed that and reported that as a bug I would say "told you so" at that point that it is no good design at all...

Reply:
Seems like this is still the case...
If no one else can fix this, I think I might give it a shot to get this done in the next few weeks.